app/frontend/src/components/form/form.test.js

```diff
@@ -1,64 +1,101 @@
 /**
  * @jest-environment jsdom
  */
-
-import form from './form';
+import form, { initAutoSubmit, initClearForm } from './form';
 
 describe('form', () => {
-  document.body.innerHTML = '<div class="clear-form"> <input id="test-text-input" type="text"/> <input id="test-checkbox" type="checkbox"/> </div>';
-  const textField = document.querySelector('#test-text-input');
-  textField.value = '01';
-  const fields = Array.from(document.querySelectorAll('input[type="text"]'));
+  describe('form auto submit behaviour', () => {
+    beforeEach(() => {
+      document.body.innerHTML = `<form data-auto-submit="true">
+      <input type="checkbox" class="govuk-checkboxes__input" data-change-submit="false" id="no-submit" />
+      <input type="checkbox" class="govuk-checkboxes__input" id="should-submit" />
+      </form>`;
+    });
+
+    describe('form has auto submit data attribute', () => {
+      let formSubmitMock = null;
+
+      beforeEach(() => {
+        form.formSubmit = jest.fn();
+        formSubmitMock = jest.spyOn(form, 'formSubmit');
+        initAutoSubmit();
+      });
+
+      test('change a input with change submit false attribute does not submit form', () => {
```

Comment:
I can't understand the sentence, is a word missing or wrong?

Reply:
yes makes no sense at all

Reply:
should be better now